### PR TITLE
AMQNET-616: Add TCP Keep-Alive support

### PR DESCRIPTION
**Pull request by Havret**
This PR adds TCP Keep-Alive support.

### CODE REVIEW

File: src/NMS.AMQP/Transport/SecureTransportContext.cs [hunk 1]
```diff
@@ -37,8 +37,8 @@ namespace Apache.NMS.AMQP.Transport
     internal class SecureTransportContext : TransportContext, ISecureTransportContext
     {
         
-        private readonly static List<string> SupportedProtocols;
-        private readonly static Dictionary<string, int> SupportedProtocolValues;
+        private static readonly List<string> SupportedProtocols;
```
Comment (michaelpearce-gain) [inline src/NMS.AMQP/Transport/SecureTransportContext.cs]: why change to static? isnt that dangerous as what occurs if multiple CF's or contexts needing different values....

Comment (HavretGC) [reply src/NMS.AMQP/Transport/SecureTransportContext.cs]: It was static from the beginning. I've merely rearranged words order to adjust it to current code style. From what I see from the code it's static as it never changes. Some consts that are initialized in static constructor.

Comment (michaelpearce-gain) [reply src/NMS.AMQP/Transport/SecureTransportContext.cs]: ah i didnt spot that

**Comment by Havret** [comment]
@cjwmorgan-sol Could you please take a look at this PR?

**Pull request merged**